xarray/tests/test_coding.py

```diff
@@ -97,20 +96,22 @@ def test_coder_roundtrip() -> None:
 
 
 @pytest.mark.parametrize("dtype", "u1 u2 i1 i2 f2 f4".split())
-def test_scaling_converts_to_float32(dtype) -> None:
+def test_scaling_converts_to_float64(dtype) -> None:
     original = xr.Variable(
         ("x",), np.arange(10, dtype=dtype), encoding=dict(scale_factor=10)
     )
     coder = variables.CFScaleOffsetCoder()
     encoded = coder.encode(original)
-    assert encoded.dtype == np.float32
+    if dtype in ["f2", "f4"]:
+        assert encoded.dtype == np.float32
+    if dtype in ["u1", "u2", "i1", "i2"]:
+        assert encoded.dtype == np.float64
     roundtripped = coder.decode(encoded)
-    assert_identical(original, roundtripped)
-    assert roundtripped.dtype == np.float32
+    assert roundtripped.dtype == np.float64
 
 
-@pytest.mark.parametrize("scale_factor", (10, [10]))
```

Comment:
I'm not sure that `scale_factor` or `add_offset` can be an array type per the CF spec, so I changed this test.

Reply:
These kinds of things tend to happen though. Since we have tested for it, we should just keep it around.